network/betweenness.go

```diff
@@ -87,3 +93,57 @@ func Betweenness(g graph.Graph) map[int]float64 {
 
 	return cb
 }
+
+// BetweennessWeighted returns the betweenness centrality for nodes in the weighted
+// graph g used to construct the given shortest paths.
+//
+//  C_B(v) = \sum_{s ≠ v ≠ t ∈ V} (\sigma_{st}(v) / \sigma_{st})
+//
+func BetweennessWeighted(g graph.CostGraph, p search.ShortestPaths) map[int]float64 {
+	nodes := g.NodeList()
+	cb := make(map[int]float64, len(nodes))
```

Comment:
I've remembered what I was thinking. I was following what I had written for the doc comment "returns the betweenness centrality for nodes in the unweighted graph g". This is unqualified. If we changed it, we'd change this to "returns the betweenness centrality for nodes in the unweighted graph g with a non-zero betweenness centrality" (and I'd need to check that this is true for Brandes' algorithm - it looks like it).

I think that we can leave this with the zero's set for a while and see how we go (I've made them match).